tests/security/ima/ima_kernel_cmdline_hash.pm

```diff
@@ -0,0 +1,82 @@
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Test IMA kernel command line for IMA hash
+# Maintainer: wnereiz <wnereiz@member.fsf.org>
+# Tags: poo#48932
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use bootloader_setup qw(add_grub_cmdline_settings replace_grub_cmdline_settings);
+use power_action_utils "power_action";
+use version_utils;
+
+sub run {
+    my ($self) = @_;
+    $self->select_serial_terminal;
+
+    my $meas_file = "/sys/kernel/security/ima/ascii_runtime_measurements";
+
+    my @algo_list = (
+        {algo => "md5",    len => 32},
+        {algo => "sha1",   len => 40},
+        {algo => "sha256", len => 64},
+        {algo => "sha512", len => 128},
+        {algo => "rmd160", len => 40},
+        {algo => "wp512",  len => 128},
+        {algo => "tgr192", len => 48},
+    );
+
+    # Add kernel modules to enable some algorithms
+    my $algo_modlist = "rmd160 wp512 tgr192";
+    $algo_modlist .= " sha512" if (!is_sle && !is_leap);
+
+    assert_script_run "echo -e $algo_modlist | sed 's/ /\\n/g' > /etc/modules-load.d/hash.conf";
+    assert_script_run "echo 'force_drivers+=\"$algo_modlist\"' >/etc/dracut.conf.d/10.hashs.conf";
+    assert_script_run "dracut -f";
+
+    add_grub_cmdline_settings('ima_policy=tcb ima_hash=none');
+    my $last_algo = "none";
+
+    for my $a (@algo_list) {
+        replace_grub_cmdline_settings("ima_hash=$last_algo", "ima_hash=@$a{algo}", 1);
+        $last_algo = @$a{algo};
+
+        my $grub_tmpfile = "/tmp/grub-" . @$a{algo};
+        assert_script_run("cp /etc/default/grub $grub_tmpfile");
+        upload_logs($grub_tmpfile);
```

Comment:
There are so many of $grub_tmpfile here so uploading all of them might not be very reasonable since we only interest in the new settings of "GRUB_CMDLINE_LINUX_DEFAULT".
We can use "diff ..." or "grep ..." to get the changes of ""GRUB_CMDLINE_LINUX_*"." and output them to "autoinst-log.txt" file.
But it is optional and not a big deal.

Others LGTM.